runs/comparisons/decision_tree.py

```diff
@@ -0,0 +1,107 @@
+from sklearn.model_selection import train_test_split
+from sklearn.utils import Bunch
+from skopt.space import Integer, Real
+
+from experiments import Experiment
+from experiments.evaluation import CrossValidateTest
+from experiments.mlflow import log_experiment
+from experiments.parameter_search import param_space
+from experiments.parameter_search.optuna import OptunaTuner
+from experiments.parameter_search.skopt import SkoptTuner
+from problems import scale_X_y
+from problems.datasets import load_airfoil_self_noise
+import click
+
+
+@click.command()
+@click.option('-p', '--problem', type=click.STRING, default='airfoil_self_noise')
+@click.option('-j', '--job_id', type=click.STRING, default='NA')
+def run(problem: str, job_id: str):
+    random_state = 42
+
+    X, y = load_airfoil_self_noise()
+    X, y = scale_X_y(X, y)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=random_state)
+
+    estimator = DecisionTreeRegressor(random_state=random_state)
+
+    default_tuner = SkoptTuner(
+        estimator=estimator,
+        X_train=X_train,
+        y_train=y_train,
+        random_state=random_state,
+        n_calls=1000,
+        verbose=10,
+        n_jobs_cv=4,
+    )
+
+    optuna_tuner = OptunaTuner(
+        estimator=estimator,
+        X_train=X_train,
+        y_train=y_train,
+        random_state=random_state,
+        n_calls=1000,
+        verbose=10,
+        n_jobs_cv=4,
+    )
+
+    # Create the base experiment, using some default tuner
+    experiment_name = f'Decision Tree {job_id} {problem}'
+    experiment = Experiment(name=experiment_name,   tuner=default_tuner, verbose=10)
+
+    # Add global tuning of the `n_estimators` parameter using optuna.
+    # It is tuned by itself first, and afterwards, the fixed value is propagated to nested experiments,
+    # because `propagate` is not set. Note that optuna does not support merging parameter spaces.
+
+    @param_space()
+    def optuna_objective(trial: optuna.Trial, params: Bunch):
+
+        params.criterion = trial.suggest_categorical(
+            'criterion', ["squared_error", "friedman_mse", "absolute_error"])
+
+        params.splitter = trial.suggest_categorical(
+            'splitter', ["best", "random"])
+
+        # params.min_samples_split = trial.suggest_int('min_samples_split', 2, 10)
+        # params.min_samples_leaf = trial.suggest_int('min_samples_leaf', 1, 4)
+        # params.min_weight_fraction_leaf = trial.suggest_float('min_weight_fraction_leaf', 0.0, 0.5)
+        params.max_features = trial.suggest_categorical('max_features', ["auto", "sqrt", "log2"])
+        # params.min_impurity_decrease = trial.suggest_float('min_impurity_decrease', 0.0, 0.5)
+        # params.ccp_alpha = trial.suggest_float('ccp_alpha', 0.0, 0.02)
+
+    experiment.with_tuning(optuna_objective, tuner=optuna_tuner)
+
+    # Create a nested experiment using the MAE.
+    mae_experiment = experiment.with_params({'criterion': 'absolute_error'})
```

Comment:
Why do we want to so this?

Reply:
I don't know the reason behind this. I just moved this file and didn't change anything there yet. I can investigate this in a different PR

Reply:
We should at least find out what it does